src/transformers/generation/configuration_utils.py

```diff
@@ -109,6 +109,17 @@ class GenerationConfig(PushToHubMixin):
         typical_p (`float`, *optional*, defaults to 1.0):
             The amount of probability mass from the original distribution to be considered in typical decoding. If set
             to 1.0 it takes no effect. See [this paper](https://arxiv.org/pdf/2202.00666.pdf) for more details.
+        epsilon_cutoff (`float`, *optional*, defaults to 0):
+            If set to float strictly between 0 and 1, only tokens with a conditional probability greater than
+            `epsilon_cutoff` will be sampled. In the paper, suggested values range from 3e-4 to 9e-4, depending on the
+            size of the model. See [Truncation Sampling as Language Model
+            Desmoothing](https://arxiv.org/abs/2210.15191) for more details.
+        eta_cutoff (`float`, *optional*, defaults to 0):
+            Eta sampling is a hybrid of locally typical sampling and epsilon sampling. If set to float strictly between
+            0 and 1, a token is only considered if it is greater than either the cutoff or sqrt(eta_cutoff) *
+            expected_next_random_token_probability. In the paper, suggested values range from 3e-4 to 2e-3, depending
```

Comment:
```suggestion
            0 and 1, a token is only considered if it is greater than either eta_cutoff or sqrt(eta_cutoff) *
            entropy(softmax(next_token_logits)). In the paper, suggested values range from 3e-4 to 2e-3, depending
```

Adding some precision to the terms here. "expected_next_random_token_probability" isn't exactly what the paper uses, if I got it right :) (feel free to correct me if you think I'm wrong!)

Reply:
It's actually `exp(-entropy(softmax(next_token_logits)))`, but fixed!